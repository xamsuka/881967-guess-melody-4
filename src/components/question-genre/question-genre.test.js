import React from 'react';
import renderer from 'react-test-renderer';
import QuestionGenreScreen from './question-genre';

const question = {
  type: `genre`,
  genre: `rock`,
  audioSrc: [
    {
      genre: `rock`,
      src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_541-fugue.ogg`
    },
    {
      genre: `rock`,
      src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_542-fugue.ogg`
    },
    {
      genre: `rock`,
      src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`
    },
    {
      genre: `rock`,
      src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_544-fugue.ogg`
    }
  ]
};

test(`<QuestionGenreScreen /> render`, () => {
  const three = renderer
    .create(<QuestionGenreScreen question = {question} onAnswerButtonSubmit ={() => {}} />).toJSON();

  expect(three).toMatchSnapshot();
});
