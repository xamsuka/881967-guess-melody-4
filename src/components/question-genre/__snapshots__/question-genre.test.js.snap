// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuestionGenreScreen /> render 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Выберите инди-рок треки
  </h2>
  <form
    className="game__tracks"
    onSubmit={[Function]}
  >
    <div
      className="track"
    >
      <button
        className="track__button track__button--play"
        onClick={[Function]}
        type="button"
      />
      <div
        className="track__status"
      >
        <audio
          src="https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_541-fugue.ogg"
        />
      </div>
      <div
        className="game__answer"
      >
        <input
          className="game__input visually-hidden"
          id="answer-0"
          name="answer"
          type="checkbox"
          value="answer-0"
        />
        <label
          className="game__check"
          htmlFor="answer-0"
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <button
        className="track__button track__button--play"
        onClick={[Function]}
        type="button"
      />
      <div
        className="track__status"
      >
        <audio
          src="https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_542-fugue.ogg"
        />
      </div>
      <div
        className="game__answer"
      >
        <input
          className="game__input visually-hidden"
          id="answer-1"
          name="answer"
          type="checkbox"
          value="answer-1"
        />
        <label
          className="game__check"
          htmlFor="answer-1"
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <button
        className="track__button track__button--play"
        onClick={[Function]}
        type="button"
      />
      <div
        className="track__status"
      >
        <audio
          src="https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg"
        />
      </div>
      <div
        className="game__answer"
      >
        <input
          className="game__input visually-hidden"
          id="answer-2"
          name="answer"
          type="checkbox"
          value="answer-2"
        />
        <label
          className="game__check"
          htmlFor="answer-2"
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <button
        className="track__button track__button--play"
        onClick={[Function]}
        type="button"
      />
      <div
        className="track__status"
      >
        <audio
          src="https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_544-fugue.ogg"
        />
      </div>
      <div
        className="game__answer"
      >
        <input
          className="game__input visually-hidden"
          id="answer-3"
          name="answer"
          type="checkbox"
          value="answer-3"
        />
        <label
          className="game__check"
          htmlFor="answer-3"
        >
          Отметить
        </label>
      </div>
    </div>
    <button
      className="game__submit button"
      type="submit"
    >
      Ответить
    </button>
  </form>
</section>
`;
